openfisca_us/variables/household/demographic/geographic/county_str.py

```diff
@@ -9,4 +9,27 @@ class county_str(Variable):
     definition_period = ETERNITY
 
     def formula(household, period, parameters):
-        return household("county", period).decode_to_str()
+        county_name = household("county", period).decode_to_str()
+        unknown_county = county_name == "UNKNOWN"
+
+        if any(unknown_county):
+            # We don't have full counties - randomly assign zip codes
+            population_data = parameters(
+                period
+            ).demographic.geography.population_by_county
+            np.random.seed(0)
+            states = np.array(list(population_data._children.keys()))
+            state = household("state_code", period)
+            for possible_state in states:
+                in_state = state == possible_state
+                possible_counties = np.array(list(population_data[possible_state]._children.keys()))
+                populations = np.array(list(population_data[possible_state]._children.values()))
+                county_name[in_state & unknown_county] = (
```

Comment:
Might be slightly more efficient (can delete `in_state` above)
```suggestion
                target = (state == possible_state) & unknown_county
                county_name[target] = (
                    np.random.choice(
                        possible_counties,
                        size=target.sum(),
                        p=populations / populations.sum(),
                    )
                )
```

Reply:
Thanks, now obsolete as this function is now a one liner and the imputation happens at the ZIP code level